### PR TITLE
You can cmag grenades to boomerang back to you

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
You can cmag grenades to boomerang back to you!

Can't get more clown than this

## Images of changes
image

## Yes I see that the bomb is doubled I'm fixing the bug in a different PR because it's a tweak and OOS

image

## Testing
See above

## Changelog
:cl:
tweak: You can now cmag grenades to boomerang back to you
/:cl:

### CODE REVIEW

**Comment by DGamerL** [comment]
> If you changed it from just syndiebombs to all grenades, shouldn't the cl be updated for that?

Good call, done!

File: code/game/objects/items/weapons/grenades/grenade.dm [hunk 1]
```diff
@@ -115,3 +114,7 @@
 	///We need to clear the walk_to on destroy to allow a grenade which uses walk_to or related to properly GC
 	walk_to(src, 0)
 	return ..()
+
+/obj/item/grenade/cmag_act(mob/user)
+	to_chat(user, "<span class='warning'>You drip some yellow ooze into [src]. [src] suddenly doesn't want to leave you...</span>")
+	AddComponent(/datum/component/boomerang, throw_range, TRUE)
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/grenades/grenade.dm:120]: is there a procedure for un cmagging a grenade?

**Comment by Scribble-Sheep** [comment]
does this proc for unarmed grenades, or only armed ones?

**Comment by DGamerL** [comment]
> does this proc for unarmed grenades, or only armed ones?

Both!

File: code/game/objects/items/weapons/grenades/grenade.dm (hunk 1 above)
Comment (DGamerL) [reply code/game/objects/items/weapons/grenades/grenade.dm:120]: Not yet, I'll add it!

**Review by lewcc — CHANGES_REQUESTED** [review]
Please add the un-cmagging behavior!

**Review by Sirryan2002 — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

**Pull request merged**